File: src/main/java/com/gmail/borlandlp/minigamesdtools/gui/other/radar2d/Radar.java
package com.gmail.borlandlp.minigamesdtools.gui.other.radar2d;

import org.bukkit.block.Block;
import org.bukkit.entity.Entity;
import org.bukkit.entity.Player;

public interface Radar {
    void draw();
    int getDrawDistance();
    void setDrawDistance(int drawDistance);
    Player getViewer();
    void setViewer(Player player);
    void addMarker(Marker marker);
    void removeMarker(Marker marker);
    Marker getCurrentMarker(Block block);
    Marker getCurrentMarker(Entity entity);
    void onUnload();
    void onLoad();
}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/gui/other/radar2d/Radar.kt
package com.gmail.borlandlp.minigamesdtools.gui.other.radar2d

import org.bukkit.block.Block
import org.bukkit.entity.Entity
import org.bukkit.entity.Player

interface Radar {
    fun draw()
    var drawDistance: Int
    var viewer: Player?
    fun addMarker(marker: Marker)
    fun removeMarker(marker: Marker)
    fun getCurrentMarker(block: Block): Marker?
    fun getCurrentMarker(entity: Entity): Marker?
    fun onUnload()
    fun onLoad()
}